### PR TITLE
Fixed secretRefs check

### PR DESCRIPTION
**Pull request by tuteng**
### Motivation

Fixed exception: 
```
client.go:98: [debug] creating 35 resource(s)
Error: PulsarBroker.pulsar.streamnative.io "pulsar-poc-sn-platform" is invalid: spec.pod.secretRefs: Invalid value: "null": spec.pod.secretRefs in body must be of type array: "null"
helm.go:75: [debug] PulsarBroker.pulsar.streamnative.io "pulsar-poc-sn-platform" is invalid: spec.pod.secretRefs: Invalid value: "null": spec.pod.secretRefs in body must be of type array: "null"
```

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: charts/sn-platform/templates/proxy/proxy-cluster.yaml [hunk 1]
```diff
@@ -57,8 +57,8 @@ spec:
 {{ toYaml . | indent 6 }}
 {{- end }}
 {{- end }}
+{{- if or .Values.proxy.readPublicKeyFromFile .Values.proxy.extraSecretRefs }}
     secretRefs:
-    {{- if .Values.proxy.readPublicKeyFromFile }}
     {{- if .Values.proxy.publicKeyPath }}
```
Comment (jdmaguire) [inline charts/sn-platform/templates/proxy/proxy-cluster.yaml:62]: I've experienced bugs like this before. The root cause of not noticing them is usually that the code isn't clear.

For example, here are five lines (if, template, else, template, end) that could be this one line:

```suggestion
    - mountPath: {{ default "/pulsar/vault/v1/identity/oidc/.well-known/keys" .Values.proxy.publicKeyPath }}
```

The lines below this and the broker-cluster.yaml (in two spots) could similarly be cleaned up.

Again, my motivation being that simpler templating leads to cleaner templating. Cleaner templating makes writing bug-free templates easier.

Comment (jdmaguire) [inline charts/sn-platform/templates/proxy/proxy-cluster.yaml:62]: I've experienced bugs like this before. The root cause of not noticing them is usually that the code isn't clear.

For example, here are five lines (if, template, else, template, end) that could be this one line:

```
- mountPath: {{ default "/pulsar/vault/v1/identity/oidc/.well-known/keys" .Values.proxy.publicKeyPath }}
```

The lines below this and the broker-cluster.yaml (in two spots) could similarly be cleaned up.

Again, my motivation being that simpler templating leads to cleaner templating. Cleaner templating makes writing bug-free templates easier.

Comment (jdmaguire) [reply charts/sn-platform/templates/proxy/proxy-cluster.yaml:62]: I do like your fix and would approve it as-is but I think we should always take the opportunity to cleanup code when it arrives and is minor.

Comment (tuteng) [reply charts/sn-platform/templates/proxy/proxy-cluster.yaml:62]: Thanks, look forward to your changes!

**Comment by Huanli-Meng** [comment]
@guangning, could you please help to verify whether we need to add/update docs for this PR? Thanks.

**Comment by tuteng** [comment]
> @Guangning, could you please help to verify whether we need to add/update docs for this PR? Thanks.

I don't think this change needs documentation

**Review by jdmaguire — APPROVED** [review]

**Pull request merged**